src/sentry/migrations/0024_auto_20191230_2052.py

```diff
@@ -87,6 +87,7 @@ def _attach_related(_events):
                 is_new_group_environment=False,
                 primary_hash=primary_hash,
                 received_timestamp=event.data.get("received")
+                or event.data.get("timestamp")
                 or float(event.datetime.strftime("%s")),
```

Comment:
I wonder if this should now come from the postgres events table rather than nodestore, i.e.

```suggestion
                or float(e.datetime.strftime("%s")),
```

This way we are looking for a valid timestamp in each of the following locations in this order:
1. Nodestore "received" field
2. Nodestore "timestamp" field
3. "Datetime" field in events table

As to why some of these may be missing for some people, I am not 100% sure, since they should all be present. I'm wondering if people who are seeing this issue somehow are totally missing nodedata for the event, (i.e. event.data is completely empty). Or if the event format is subtly different in some old (pre v9 and possibly much earlier) versions of Sentry which they are attempting to upgrade from.

Reply:
> I wonder if this should now come from the postgres events table rather than nodestore, i.e.

yeah, the issue I faced was from the install that use postgres. As for the version, i was trying to upgrade from 9.1.2 when faced an issue and tried to fix;  meanwhile, the installation age is about 3 years, I was installing updates just via pip, so there could be some "cumulative" errors